Assets/Game/Addons/WorldDataEditor/Editor/WorldDataEditor.cs

```diff
@@ -0,0 +1,1501 @@
+using System.Collections.Generic;
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+using DaggerfallConnect;
+using DaggerfallConnect.Arena2;
+using DaggerfallWorkshop.Utility.AssetInjection;
+
+namespace DaggerfallWorkshop.Game.Utility.WorldDataEditor
+{
+    /// <summary>
+    /// WorldData editor
+    /// </summary>
+    public class WorldDataEditor : EditorWindow
+    {
+        public const string WorldDataFolder = "/StreamingAssets/WorldData/";
+        private GUIStyle lightBG = new GUIStyle();
+        private GUIStyle lightMedBG = new GUIStyle();
+        private GUIStyle bigText = new GUIStyle();
+
+        private void CreataGUIStyles()
+        {
+            if (EditorGUIUtility.isProSkin)
+            {
+                lightBG.normal.background = WorldDataEditorBuildingHelper.CreateColorTexture(1, 1, new Color(0.35f, 0.35f, 0.35f, 1.0f));
+                lightMedBG.normal.background = WorldDataEditorBuildingHelper.CreateColorTexture(1, 1, new Color(0.25f, 0.25f, 0.25f, 1.0f));
+            }
+            else
+            {
+                lightBG.normal.background = WorldDataEditorBuildingHelper.CreateColorTexture(1, 1, new Color(0.98f, 0.98f, 0.98f, 1.0f));
```

Comment:
Am I the only one who finds these backgrounds completely unreadable in UV's original editor? Maybe that's what the "pro skin" check is about, I'm not sure if my editor is pro skin or not

Reply:
Yeah, it checks if you're using dark UI (pro) and adjusts colours accordingly. I didn't bother allowing it to be switched on the fly though.